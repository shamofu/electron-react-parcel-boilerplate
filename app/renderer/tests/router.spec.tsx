import React from 'react';
import Renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Router from '../router';

test('route /', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={['/']}><Router /></MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('route /start', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={['/start']}><Router /></MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('route /dummy', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={['/dummy']}><Router /></MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
