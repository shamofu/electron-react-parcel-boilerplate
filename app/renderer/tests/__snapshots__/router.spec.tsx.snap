// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route / 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <h3>
    <a
      href="/start"
      id="start"
      onClick={[Function]}
    >
      Start
    </a>
  </h3>
</div>
`;

exports[`route /dummy 1`] = `
<h1>
  204 No Content
</h1>
`;

exports[`route /start 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <h1>
    Electron + React + Parcel
  </h1>
  <h3>
    <a
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </h3>
</div>
`;
